mnist-dp/README.md

```diff
@@ -0,0 +1,199 @@
+> [Differential privacy](https://en.wikipedia.org/wiki/Differential_privacy) (DP) is a framework for measuring the privacy guarantees provided by an algorithm. Through the lens of differential privacy, we can design machine learning algorithms that responsibly train models on private data. Learning with differential privacy provides provable guarantees of privacy, mitigating the risk of exposing sensitive training data in machine learning. Intuitively, a model trained with differential privacy should not be affected by any single training example, or small set of training examples, in its data set. This mitigates the risk of exposing sensitive training data in ML.
+>  
+> The basic idea of this approach, called differentially private stochastic gradient descent (DP-SGD), is to modify the gradients used in stochastic gradient descent (SGD), which lies at the core of almost all deep learning algorithms. Models trained with DP-SGD provide provable differential privacy guarantees for their input data. There are two modifications made to the vanilla SGD algorithm:
+>  
+> 1. First, the sensitivity of each gradient needs to be bounded. In other words, we need to limit how much each individual training point sampled in a minibatch can influence gradient computations and the resulting updates applied to model parameters. This can be done by *clipping* each gradient computed on each training point.
+> 2. *Random noise* is sampled and added to the clipped gradients to make it statistically impossible to know whether or not a particular data point was included in the training dataset by comparing the updates SGD applies when it operates with or without this particular data point in the training dataset.
+>  
+> This tutorial uses [tf.keras](https://www.tensorflow.org/guide/keras) to train a convolutional neural network (CNN) to recognize handwritten digits with the DP-SGD optimizer provided by the TensorFlow Privacy library. TensorFlow Privacy provides code that wraps an existing TensorFlow optimizer to create a variant that implements DP-SGD.  
+> &mdash; [The TensorFlow Authors][1]
+
+[1]: https://github.com/tensorflow/privacy/blob/master/tutorials/Classification_Privacy.ipynb
+
+## Prerequisites
+
+In order to run this example, you'll need to:
+
+* use Python 3
+* have [Docker](https://www.docker.com/) installed
+* [install the `substra` cli](https://github.com/SubstraFoundation/substra#install) (supported version: 0.6.0)
+* [install the `substratools` library](https://github.com/substrafoundation/substra-tools) (supported version: 0.6.0)
+* [pull the `substra-tools` docker images](https://github.com/substrafoundation/substra-tools#pull-from-private-docker-registry)
+* have access to a Substra installation ([configure your host to a public node ip](https://doc.substra.ai/getting_started/installation/local_install_skaffold.html#network) or [install Substra on your machine](https://doc.substra.ai/getting_started/installation/local_install_skaffold.html))
```

Comment:
In fact it would be great to add a reference to the compatibility table here to ensure that it works with the cli 0.6.0: https://github.com/SubstraFoundation/substra#compatibility-table